ci/integration-tests/src/helpers/chain.rs

```diff
@@ -0,0 +1,126 @@
+use cosm_orc::config::key::Key;
+use cosm_orc::orchestrator::cosm_orc::CosmOrc;
+use cosm_orc::{config::cfg::Config, config::key::SigningKey};
+use once_cell::sync::OnceCell;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::path::Path;
+use std::time::Duration;
+use test_context::TestContext;
+
+static CONFIG: OnceCell<Cfg> = OnceCell::new();
+
+#[derive(Debug)]
+pub struct Cfg {
+    cfg: Config,
+    gas_report_dir: String,
+}
+
+pub struct Chain {
+    pub cfg: Config,
+    pub orc: CosmOrc,
+    pub users: HashMap<String, SigningAccount>,
+}
+
+#[derive(Clone, Debug)]
+pub struct SigningAccount {
+    pub account: Account,
+    pub key: SigningKey,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Account {
+    pub name: String,
+    pub address: String,
+    pub mnemonic: String,
+}
+
+// NOTE: we have to run the integration tests in one thread right now.
+// We get `account sequence mismatch` CosmosSDK error when we run in parallel.
+// We could either serialize the `account.sequence` per key, or use a different key per test.
+
+impl TestContext for Chain {
+    fn setup() -> Self {
+        let cfg = CONFIG.get_or_init(global_setup).cfg.clone();
+        let orc = CosmOrc::new(cfg.clone(), true).unwrap();
+        let users = test_accounts();
+        Self { cfg, orc, users }
+    }
+
+    fn teardown(self) {
+        let cfg = CONFIG.get().unwrap();
+        save_gas_report(&self.orc, &cfg.gas_report_dir);
+    }
+}
+
+fn test_accounts() -> HashMap<String, SigningAccount> {
+    let bytes = fs::read("../configs/test_accounts.json").unwrap();
+    let accounts: Vec<Account> = serde_json::from_slice(&bytes).unwrap();
+
+    let mut account_map = HashMap::new();
+    for account in accounts {
+        account_map.insert(
+            account.name.clone(),
+            SigningAccount {
+                account: account.clone(),
+                key: SigningKey {
+                    name: account.name,
+                    key: Key::Mnemonic(account.mnemonic),
+                },
+            },
+        );
+    }
+    account_map
```

Comment:
nit: rusty way :sunglasses: 
```suggestion
    accounts
        .into_iter()
        .map(|account| {
            (
                account.name.clone(),
                SigningAccount {
                    account: account.clone(),
                    key: SigningKey {
                        name: account.name,
                        key: Key::Mnemonic(account.mnemonic),
                    },
                },
            )
        })
        .collect::<HashMap<String, SigningAccount>>()
```